commit/chainfee/outcome.go

```diff
@@ -0,0 +1,185 @@
+package chainfee
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/mathslib"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon/consensus"
+)
+
+func (p *processor) getConsensusObservation(
+	aos []plugincommon.AttributedObservation[Observation],
+) (Observation, error) {
+	aggObs := aggregateObservations(aos)
+
+	// consensus on the fChain map uses the role DON F value
+	// because all nodes can observe the home chain.
+	donThresh := consensus.MakeConstantThreshold[cciptypes.ChainSelector](consensus.TwoFPlus1(p.fRoleDON))
+	fChains := consensus.GetConsensusMap(p.lggr, "fChain", aggObs.FChain, donThresh)
+
+	fDestChain, exists := fChains[p.destChain]
+	if !exists {
+		return Observation{},
+			fmt.Errorf("no consensus value for fDestChain, destChain: %d", p.destChain)
+	}
+	if len(aggObs.Timestamps) < fDestChain {
+		return Observation{},
+			fmt.Errorf("not enough observations for timestamps to reach consensus, have %d, need %d",
+				len(aggObs.Timestamps), fDestChain)
+	}
+	timestamp := consensus.Median(aggObs.Timestamps, consensus.TimestampComparator)
+
+	chainFeeUpdatesConsensus := consensus.GetConsensusMapAggregator(
+		p.lggr,
+		"ChainFeeUpdates",
+		aggObs.ChainFeeUpdates,
+		consensus.MakeConstantThreshold[cciptypes.ChainSelector](consensus.TwoFPlus1(fDestChain)),
+		plugincommon.ChainFeeUpdateAggregator,
+	)
+
+	twoFPlus1 := consensus.MakeMultiThreshold(fChains, consensus.TwoFPlus1)
```

Comment:
nit: naming for clarity

```suggestion
	twoFChainPlus1 := consensus.MakeMultiThreshold(fChains, consensus.TwoFPlus1)
```